gpytorch/kernels/kernel.py

```diff
@@ -183,13 +183,28 @@ def _create_input_grid(self, x1, x2):
             The shape depends on the kernel's mode
 
             * `full_covar`: (`b x n x 1 x d` and `b x 1 x m x d`)
-            * `full_covar` with `dim_groups=k`: (`b x k x n x 1 x (d//k)` and `b x k x 1 x m x (d//k)`)
+            * `full_covar` with `batch_dims=(0, 2)`: (`b x k x n x 1 x (d//k)` and `b x k x 1 x m x (d//k)`)
             * `diag`: (`b x n x d` and `b x n x d`)
-            * `diag` with `dim_groups=k`: (`b x k x n x (d//k)` and `b x k x n x (d//k)`)
+            * `diag` with `batch_dims=(0, 2)`: (`b x k x n x (d//k)` and `b x k x n x (d//k)`)
         """
-        return x1.unsqueeze(-2), x2.unsqueeze(-3)
+        x1_, x2_ = x1, x2
+        if batch_dims == (0, 2):
```

Comment:
again should we raise a NotImplementedError here if we get values that we don't yet support?

Reply:
At the moment the check in __call__ will raise an error if it's not something we support